bugshaker/src/main/java/com/github/stkent/bugshaker/flow/email/FeedbackEmailFlowManager.java

```diff
@@ -30,221 +35,270 @@
+	}
+
+	private void showDialog(Context applicationContext) {
+
+		final Activity currentActivity = activityReferenceManager.getValidatedActivity();
+		if (currentActivity == null) {
+			return;
+		}
+
+		AlertDialog.Builder bugAlertBuilder = new AlertDialog.Builder(currentActivity);
+		bugAlertBuilder.setMessage(applicationContext.getString(R.string.shake_detected) + '\n' + applicationContext
+			.getString(R.string.report_a_bug));
+		bugAlertBuilder.setCancelable(false);
+
+		bugAlertBuilder.setPositiveButton(applicationContext.getString(R.string.report), reportBugClickListener);
+		bugAlertBuilder.setNegativeButton(applicationContext.getString(R.string.cancel), null);
+		bugAlertBuilder
+			.setNeutralButton(applicationContext.getString(R.string.annotate_and_report), screenshotListener);
+
+		AlertDialog alertDialog = bugAlertBuilder.create();
+		alertDialog.show();
+
+	}
+
+	private void dismissDialog() {
+		if (alertDialog != null) {
+			alertDialog.dismiss();
+			alertDialog = null;
+		}
+	}
+
+	private boolean shouldAttemptToCaptureScreenshot(@NonNull final Activity activity) {
+		final int windowFlags = ActivityUtils.getWindow(activity).getAttributes().flags;
+
+		final boolean isWindowSecured =
+			(windowFlags & WindowManager.LayoutParams.FLAG_SECURE) == FLAG_SECURE_VALUE;
+
+		final boolean result = ignoreFlagSecure || !isWindowSecured;
+		final Logger logger = new Logger(true);
+
+		if (!isWindowSecured) {
+			logger.d("Window is not secured; should attempt to capture screenshot.");
+		}
+		else {
+			if (ignoreFlagSecure) {
+				logger.d("Window is secured, but we're ignoring that.");
+			}
+			else {
+				logger.d("Window is secured, and we're respecting that.");
+			}
+		}
+
+		return result;
+	}
+
+	public void sendEmailWithScreenshot(
+		@NonNull final Context context,
+		@NonNull final Uri screenshotUri, final Uri file) {
+		final Intent feedbackEmailIntent = FeedbackEmailIntentUtil
+			.getFeedbackEmailIntent(context, emailAddresses, emailSubjectLine, screenshotUri, file);
+
+		final List<ResolveInfo> resolveInfoList = context.getPackageManager()
+			.queryIntentActivities(feedbackEmailIntent, PackageManager.MATCH_DEFAULT_ONLY);
+
+		for (final ResolveInfo receivingApplicationInfo : resolveInfoList) {
+			context.grantUriPermission(
+				receivingApplicationInfo.activityInfo.packageName,
+				screenshotUri,
+				Intent.FLAG_GRANT_READ_URI_PERMISSION);
+		}
+		Intent sendEmailIntent = Intent.createChooser(feedbackEmailIntent, "Send email");
+		sendEmailIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+		context.startActivity(sendEmailIntent);
+		LogcatUtil.getLogFile().delete();
+	}
+
+	private void sendEmailWithoutScreenshot(Context applicationContext, @NonNull final Activity activity) {
+		final Intent feedbackEmailIntent = FeedbackEmailIntentUtil
+			.getFeedbackEmailIntent(applicationContext,
+				emailAddresses, emailSubjectLine);
+		final Logger logger = new Logger(true);
+
```

Comment:
we use the logger later on to print "sending email with no screenshot" in the log, do you want us to get rid of both final Logger logger = new Logger(true) and the line logger.d("sending email with no screenshot")?